app/services/licences/fetch-licence-version-purpose-condition.service.js

```diff
@@ -10,45 +10,74 @@ const LicenceVersionPurposeCondtionModel = require('../../models/licence-version
  *
  * @returns {Promise<Object>} the data needed to populate the view licence page and some elements of the summary tab
  */
-async function go (id) {
-  if (!id) {
+async function go (ids) {
+  if (!ids) {
     return {
-      abstractionConditions: []
+      numberOfAbstractionConditions: 0,
+      uniqueAbstractionConditions: []
     }
   }
-  const LicenceVersionPurposeCondition = await _fetchLicenceVersionPurposeCondition(id)
-  const data = await _data(LicenceVersionPurposeCondition)
+
+  const licenceVersionIds = []
+
+  ids.forEach((item) => { licenceVersionIds.push(item.licenceVersionPurposeId) })
+
+  const LicenceVersionPurposeCondition = await _fetchLicenceVersionPurposeCondition(licenceVersionIds)
+  const data = await _data(LicenceVersionPurposeCondition, ids)
 
   return data
 }
 
-async function _data (licenceVersionPurposeConditions) {
+async function _data (licenceVersionPurposeConditions, licenceVersionPuroseAndPurposeUseIds) {
   const abstractionConditions = []
 
   if (!licenceVersionPurposeConditions) {
-    return abstractionConditions
+    return {
+      numberOfAbstractionConditions: 0,
+      uniqueAbstractionConditions: []
+    }
   }
 
-  licenceVersionPurposeConditions.forEach((purposeConditions) => {
-    purposeConditions.licenceVersionPurposeConditionTypes.forEach((licenceVersionPurposeConditionType) => {
-      abstractionConditions.push(licenceVersionPurposeConditionType.displayTitle)
+  licenceVersionPurposeConditions.forEach((purposeCondition) => {
+    // Get the purposeUseId from the LicenceVersionPurpose table using the licenceVersionPurposeId of the current purposeCondition
+    const { purposeId: purposeUseId } = licenceVersionPuroseAndPurposeUseIds.find((item) =>
+      item.licenceVersionPurposeId === purposeCondition.licenceVersionPurposeId)
+
+    // Loop through each of the condition types and check to see if there is already a set of those conditions in the array
+    purposeCondition.licenceVersionPurposeConditionTypes.forEach((licenceVersionPurposeConditionType) => {
+      if (!abstractionConditions.find((element) =>
+        element.displayTitle === licenceVersionPurposeConditionType.displayTitle &&
+        element.purposeUseId === purposeUseId
+      )) {
+        abstractionConditions.push({
+          displayTitle: licenceVersionPurposeConditionType.displayTitle,
+          purposeUseId
+        })
+      }
     })
   })
 
+  const abstractionConditionDisplayTitles = []
+  abstractionConditions.forEach((condition) => {
+    abstractionConditionDisplayTitles.push(condition.displayTitle)
+  })
+
   return {
-    abstractionConditions: [...new Set(abstractionConditions)]
+    numberOfAbstractionConditions: abstractionConditions.length,
+    uniqueAbstractionConditions: [...new Set(abstractionConditionDisplayTitles)]
   }
 }
 
-async function _fetchLicenceVersionPurposeCondition (id) {
+async function _fetchLicenceVersionPurposeCondition (ids) {
```

Comment:
As this returns an array rather than a single object we should probably pluralise the method name.

```suggestion
async function _fetchLicenceVersionPurposeConditions (ids) {
```